src/core/mac/channel_mask.hpp

```diff
@@ -140,7 +140,7 @@ class ChannelMask
      * @returns TRUE if the channel @p aChannel is included in the mask, FALSE otherwise.
      *
      */
-    bool ContainsChannel(uint8_t aChannel) const { return ((1U << aChannel) & mMask) != 0; }
+    bool ContainsChannel(uint8_t aChannel) const { return (aChannel < 32) ? ((1U << aChannel) & mMask) != 0 : false; }
```

Comment:
The `ChannelMask` APIs inherently assumed that `aChannel` parameter is always within correct range. 

If we change `ContainsChannel()` to check for this I suggest to check in `AddChannel` and `RemoveChannel` as well. 

Thoughts?